README.md

```diff
@@ -76,31 +77,30 @@ transformation code.
 // Toy has literals, variables, basic arithmetic and functions that accept a single argument. 
 
 (define toy_lang
-    (domain
-        (sum expr
-            (lit value::ion)
-            (variable name::symbol)
-            (not expr::expr)
-            (apply op operands::(* expr 2))
-            (call name::symbol argument::expr)
-            (let name::symbol value::expr body::expr)
-            (function var_name::symbol body::expr))
-        (sum op (plus) (minus) (times) (divide))
-))
-
-// Define another type domain which is the same as "toy_lang" but replaces variable names with DeBruijn indices:
-(define toy_lang_nameless
+ (domain
+   (sum operator (plus) (minus) (times) (divide) (modulo))
+    (sum expr
+        (lit value::ion)
+        (variable name::symbol)
+        (not expr::expr)
+        (nary op::operator operands::(* expr 0))
+        (let name::symbol value::expr body::expr)
+        (function var_name::symbol body::expr))))
+
+// Define another type domain which extends "toy_lang" by removing named variables and adding DeBruijn indices:
```

Comment:
The `name::symbol` parameters are still being defined for `variable` and `let`. The named variables are still included in `toy_lang_indexed` and aren't removed (unless I'm misinterpreting what named variables are)?

Reply:
Not sure I understand exactly here... `exclude` removes the named type and all of its properties.

Then, `include` adds a new type.  In this case, the new type has the same elements of the removed type plus the index.  That is actually intentional, but not *required* for anything.  I may need to understand what's confusing about this to suggest something that would be clearer or do you have a suggestion?

Reply:
Oh I had mistaken the changed comment on L90 "removing named variables and adding DeBruijn indices" for what was part of the diff (L96-L98). I didn't realize the comment was referring to the `exclude variable let` on L94.